x-pack/metricbeat/module/azure/azure.go

```diff
@@ -63,3 +65,63 @@ func newModule(base mb.BaseModule) (mb.Module, error) {
 	}
 	return &base, nil
 }
+
+// MetricSet holds any configuration or state information. It must implement
+// the mb.MetricSet interface. And this is best achieved by embedding
+// mb.BaseMetricSet because it implements all of the required mb.MetricSet
+// interface methods except for Fetch.
+type MetricSet struct {
+	mb.BaseMetricSet
+	Client *Client
+}
+
+// NewMetricSet will instantiate a new azure metricset
+func NewMetricSet(base mb.BaseMetricSet) (*MetricSet, error) {
+	metricsetName := base.Name()
+	cfgwarn.Beta("The azure %s metricset is beta.", metricsetName)
+	var config Config
+	err := base.Module().UnpackConfig(&config)
+	if err != nil {
+		return nil, errors.Wrap(err, "error unpack raw module config using UnpackConfig")
+	}
+
+	//validate config based on metricset
+	switch base.Name() {
```

Comment:
Could this be `metricsetName`?
```suggestion
	switch metricsetName {
```